src/node/genesisgen.h

```diff
@@ -207,7 +207,9 @@ namespace ccf
         return false;
       }
 
-      if (active_service->status != ServiceStatus::OPENING)
+      if (
+        active_service->status != ServiceStatus::OPENING &&
```

Comment:
Maybe it's worth using the state machine code we use for the node status here too, to have pre-condition checks?

Reply:
Good point. I've added an item for this on #51. Once the recovery with key shares is done, I also want to split `genesisgen.h` into separate bits for each table/group of tables so will probably do it then.